torch/distributed/fsdp/_optim_utils.py

```diff
@@ -1410,6 +1420,8 @@ def _optim_state_dict(
     to_save = not rank0_only or (dist.get_rank(group) == 0 or shard_state)
     fsdp_osd: Dict[str, Any] = {"state": {}, "param_groups": []} if to_save else {}
     fsdp_osd_state: Dict[str, Any] = fsdp_osd["state"] if to_save else {}
+    if "param_groups" not in optim_state_dict:
```

Comment:
Instead of popping out, we can change the initialization of `fsdp_osd` to only initialize `state`. The `param_groups` will be assigned later by `_unflatten_param_groups` in new line 1529.